realhf/api/from_hf/mixtral.py

```diff
@@ -0,0 +1,237 @@
+                f"{i}.attn.c_attn.k_attn.weight"
+            ]
+            new_sd[f"model.layers.{i-1}.self_attn.o_proj.weight"] = state_dict[
+                f"{i}.attn.c_proj.weight"
+            ]
+            new_sd[f"model.layers.{i-1}.self_attn.q_proj.weight"] = state_dict[
+                f"{i}.attn.c_attn.q_attn.weight"
+            ]
+            new_sd[f"model.layers.{i-1}.self_attn.v_proj.weight"] = state_dict[
+                f"{i}.attn.c_attn.v_attn.weight"
+            ]
+            if config.use_attention_bias:
+                new_sd[f"model.layers.{i-1}.self_attn.q_proj.bias"] = state_dict[
+                    f"{i}.attn.c_attn.q_attn.bias"
+                ]
+                new_sd[f"model.layers.{i-1}.self_attn.k_proj.bias"] = state_dict[
+                    f"{i}.attn.c_attn.k_attn.bias"
+                ]
+                new_sd[f"model.layers.{i-1}.self_attn.v_proj.bias"] = state_dict[
+                    f"{i}.attn.c_attn.v_attn.bias"
+                ]
+            if config.use_attn_proj_bias:
+                new_sd[f"model.layers.{i-1}.self_attn.o_proj.bias"] = state_dict[
+                    f"{i}.attn.c_proj.bias"
+                ]
+            new_sd[f"model.layers.{i-1}.self_attn.rotary_emb.inv_freq"] = 1.0 / (
+                config.rotary_base
+                ** (
+                    torch.arange(
+                        0,
+                        config.head_dim,
+                        2,
+                        device=device,
+                        dtype=torch.float32,
+                    )
+                    / config.head_dim
+                )
+            )
+            if i == config.n_layers:
+                new_sd["model.norm.weight"] = state_dict[f"{i}.ln_f.weight"]
+
+    return new_sd
+
+
+def mixtral_transformer_block_param_name(
+    config: ReaLModelConfig, idx: int
+) -> List[str]:
+    names = []
+    num_experts = config.moe.num_experts
+    for k in ["weight", "bias"]:
+        names += [
+            f"model.layers.{idx}.input_layernorm.{k}",
+            f"model.layers.{idx}.block_sparse_moe.gate.{k}",
+            f"model.layers.{idx}.post_attention_layernorm.{k}",
+            f"model.layers.{idx}.self_attn.k_proj.{k}",
+            f"model.layers.{idx}.self_attn.o_proj.{k}",
+            f"model.layers.{idx}.self_attn.q_proj.{k}",
+            # f"model.layers.{idx}.self_attn.rotary_emb.inv_freq",
+            f"model.layers.{idx}.self_attn.v_proj.{k}",
+        ]
+        for j in range(num_experts):
+            names += [
+                f"model.layers.{idx}.block_sparse_moe.experts.{j}.w1.{k}",
+                f"model.layers.{idx}.block_sparse_moe.experts.{j}.w2.{k}",
+                f"model.layers.{idx}.block_sparse_moe.experts.{j}.w3.{k}",
+            ]
+        if idx == config.n_layers - 1:
+            names += [f"model.norm.{k}"]
+    return names
+
+
+def get_real_config_mixtral() -> ReaLModelConfig:
+    hf_config = transformers.MixtralConfig(
```

Comment:
Shrink the hidden size using the global variable in `base/constants`. See other HF implementations.

Reply:
Fixed